plugin/lighthouse/setup.go

```diff
@@ -22,25 +21,51 @@ func init() {
 // setup is the function that gets called when the config parser see the token "lighthouse". Setup is responsible
 // for parsing any extra options the this plugin may have. The first token this function sees is "lighthouse".
 func setupLighthouse(c *caddy.Controller) error {
-	c.Next() // Ignore "lighthouse" and give us the next token.
-	if c.NextArg() {
-		// If there was another token, return an error, because we currently don't have any configuration.
-		// Any errors returned from this setup function should be wrapped with plugin.Error, so we
-		// can present a slightly nicer error message to the user.
-		return plugin.Error("lighthouse", c.ArgErr())
+	l, err := lighthouseParse(c)
+	if err != nil {
+		return plugin.Error("lighthouse", err)
 	}
 
-	svcsMap := setupServicesMap()
-	log.Debugf("Services Map for lighthouse is %s", svcsMap)
-	// Add the Plugin to CoreDNS, so Servers can use it in their plugin chain.
 	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
-		return &Lighthouse{Next: next, SvcsMap: svcsMap}
+		l.Next = next
+		return l
 	})
 
-	// All OK, return a nil error.
 	return nil
 }
 
+func lighthouseParse(c *caddy.Controller) (*Lighthouse, error) {
+	lh := Lighthouse{}
+	// Changed `for` to `if` to satisfy golint:
+	//	 SA4004: the surrounding loop is unconditionally terminated (staticcheck)
+	if c.Next() {
+		lh.Zones = c.RemainingArgs()
+		if len(lh.Zones) == 0 {
+			lh.Zones = make([]string, len(c.ServerBlockKeys))
+			copy(lh.Zones, c.ServerBlockKeys)
+		}
+		for i, str := range lh.Zones {
+			lh.Zones[i] = plugin.Host(str).Normalize()
+		}
+
+		for c.NextBlock() {
+			switch c.Val() {
+			case "fallthrough":
+				lh.Fall.SetZonesFromArgs(c.RemainingArgs())
+			case "clusters":
```

Comment:
Why do you add clusters config?, that may be automatically discovered, may be I'm confusing where does this go?

Reply:
This will be used to specify proximity order of clusters. If service is available in multiple clusters, this list will be used to determine which is the closest one, and should return the IP from that one.

Reply:
Removed this option. Need a different way to get cluster order from admin, will be a separate feature.